source/extensions/omni.isaac.lab/omni/isaac/lab/assets/deformable_object/deformable_object.py

```diff
@@ -0,0 +1,413 @@
+        it depends. We will try to maintain backward compatibility as much as possible but some changes may be
+        necessary.
+
+    .. _partially kinematic: https://nvidia-omniverse.github.io/PhysX/physx/5.4.1/docs/SoftBodies.html#kinematic-soft-bodies
+    """
+
+    cfg: DeformableObjectCfg
+    """Configuration instance for the deformable object."""
+
+    def __init__(self, cfg: DeformableObjectCfg):
+        """Initialize the deformable object.
+
+        Args:
+            cfg: A configuration instance.
+        """
+        super().__init__(cfg)
+
+    """
+    Properties
+    """
+
+    @property
+    def data(self) -> DeformableObjectData:
+        return self._data
+
+    @property
+    def num_instances(self) -> int:
+        return self.root_physx_view.count
+
+    @property
+    def num_bodies(self) -> int:
+        """Number of bodies in the asset.
+
+        This is always 1 since each object is a single deformable body.
+        """
+        return 1
+
+    @property
+    def root_physx_view(self) -> physx.SoftBodyView:
+        """Deformable body view for the asset (PhysX).
+
+        Note:
+            Use this view with caution. It requires handling of tensors in a specific way.
+        """
+        return self._root_physx_view
+
+    @property
+    def material_physx_view(self) -> physx.SoftBodyMaterialView | None:
+        """Deformable material view for the asset (PhysX).
+
+        This view is optional and may not be available if the material is not bound to the deformable body.
+        If the material is not available, then the material properties will be set to default values.
+
+        Note:
+            Use this view with caution. It requires handling of tensors in a specific way.
+        """
+        return self._material_physx_view
+
+    @property
+    def max_sim_mesh_elements_per_body(self) -> int:
+        """The maximum number of simulation mesh elements per deformable body."""
+        return self.root_physx_view.max_sim_elements_per_body
+
+    @property
+    def max_collision_mesh_elements_per_body(self) -> int:
+        """The maximum number of collision mesh elements per deformable body."""
+        return self.root_physx_view.max_elements_per_body
+
+    @property
+    def max_sim_mesh_vertices_per_body(self) -> int:
+        """The maximum number of simulation mesh vertices per deformable body."""
+        return self.root_physx_view.max_sim_vertices_per_body
+
+    @property
+    def max_collision_mesh_vertices_per_body(self) -> int:
+        """The maximum number of collision mesh vertices per deformable body."""
+        return self.root_physx_view.max_vertices_per_body
+
+    """
+    Operations.
+    """
+
+    def reset(self, env_ids: Sequence[int] | None = None):
+        # Think: Should we reset the kinematic targets when resetting the object?
+        #  This is not done in the current implementation. We assume users will reset the kinematic targets.
```

Comment:
Can you make a ticket for this?

Reply:
@masoudmoghani Can you please make a ticket on this? I think in general we shouldn't do any hidden operation but maybe worth keeping in mind as more people adopt deformabels.